core/interfaces/i_draggable.ts

```diff
@@ -4,11 +4,63 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-// Former goog.module ID: Blockly.IDraggable
+import {Coordinate} from '../utils/coordinate';
+import {IDragTarget} from './i_drag_target';
 
-import type {IDeletable} from './i_deletable.js';
+/**
+ * Represents an object that can be dragged.
+ */
+export interface IDraggable extends IDragStrategy {
+  /** Returns true iff the element is currently movable. */
+  isMovable(): boolean;
+
+  /** Returns the current location of the draggable in workspace coordinates. */
+  getLocation(): Coordinate;
+}
 
 /**
- * The interface for an object that can be dragged.
+ * Represents an object that can be dragged.
  */
-export interface IDraggable extends IDeletable {}
+export interface IDragStrategy {
+  /**
+   * Handles any drag startup (e.g moving elements to the front of the
+   * workspace).
+   *
+   * @param e PointerEvent that started the drag; could be used to
+   *     check modifier keys, etc.  May be missing when dragging is
+   *     triggered programatically rather than by user.
+   */
+  startDrag(e?: PointerEvent): void;
+
+  /**
+   * Handles moving elements to the new location, and updating any
+   * visuals based on that (e.g connection previews for blocks).
+   *
+   * @param newLoc Workspace coordinate to which the draggable has
+   *     been dragged.
+   * @param e PointerEvent that continued the drag; could be used to
+   *     look up any IDragTarget the pointer is over, check modifier
+   *     keys, etc.
```

Comment:
If provided, they /should/ look up the drag target and deal with it.

Reply:
Good point.